client/components/tinymce/plugins/mentions/suggestion-list.jsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenu from 'components/popover/menu';
+import PopoverMenuItem from 'components/popover/menu-item';
+import Suggestion from './suggestion';
+
+const SuggestionList = React.createClass( {
+	displayName: 'SuggestionList',
+
+	propTypes: {
+		query: React.PropTypes.string,
+		suggestions: React.PropTypes.array,
+		isVisible: React.PropTypes.bool,
+		onClick: React.PropTypes.func,
+		onClose: React.PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			query: '',
+			suggestions: [],
+			isVisible: false
+		};
+	},
+
+	getMatchingSuggestions() {
+		const { suggestions, query } = this.props;
+		const matcher = new RegExp( '^' + query + '| ' + query, 'ig' ); // Start of string or preceded by a space.
+		let matchingSuggestions = suggestions;
+
+		if ( query.length > 0 ) {
+			matchingSuggestions = [];
+
+			for ( let i = 0, len = suggestions.length; i < len; i++ ) {
+				const suggestion = suggestions[ i ];
+				const name = suggestion.name || suggestion.user_login + ' ' + suggestion.display_name;
+
+				if ( name.toLowerCase().match( matcher ) ) {
+					matchingSuggestions.push( suggestion );
+				}
+			}
+		}
+
+		return matchingSuggestions.slice( 0, 10 );
+	},
+
+	render() {
+		if ( ! this.props.isVisible || ! this.props.suggestions ) {
+			return null;
+		}
+
+		const suggestions = this.getMatchingSuggestions();
+
+		if ( suggestions.length > 0 ) {
+			return (
+				<div ref="mention">
+					<PopoverMenu
+						className="suggestions"
+						context={ this.refs && this.refs.mention }
+						isVisible={ this.props.isVisible }
+						onClose={ this.props.onClose }>
+							{ suggestions.map( ( suggestion ) => {
+								return (
+									<PopoverMenuItem
+										key={ 'user-suggestion-' + suggestion.ID }
+										onClick={ this.props.onClick.bind( null, suggestion ) }>
```

Comment:
@aduth it's not a problem with the intent of the rule, but making the linter happy won't actually accomplish the spirit of the rule. we're inside of a map and need to pass along some state inside a closure. in the case of the fat-arrow function its explicit and clear what it's doing. unfortunately, the linter throws a fit because it doesn't want us to create a new closure inside of `render()` so we end up `bind()`ing which makes the linter happy but ends up doing exactly the same thing under the hood - it creates a new closure. this is one of those cases where we have a data dependency inside of `render()` and there's no possible way to eliminate the fact that we create creating a new function here unless we elect a really inappropriate and ugly workaround, again simply to make the linter happy.

I do not believe I'm wrong here and I've seen this many times in our code. If you think I'm wrong then I will definitely need to learn why so :smile: